### PR TITLE
Signup: Add a param to hide progress indicator in flows

### PR DESCRIPTION
**Pull request by aidvu**
Added a param `hideProgressIndicator` to allow us to hide the progress indicator in certain flows.

Addresses parts of #11480.

Test:
1. Purchase domain using domain only site flow.
2. Go to My Sites.
3. Click `Create Site` button.
4. Shouldn't see the `Step # of #`.

### Review

* [ ] Code
* [ ] Product

### CODE REVIEW

File: client/signup/flow-progress-indicator/index.jsx [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
 /**
  * Internal dependencies
  */
-var flows = require( 'signup/config/flows' );
+import flows from 'signup/config/flows';
+import { localize } from 'i18n-calypso';
 
-module.exports = React.createClass( {
-	displayName: 'FlowProgressIndicator',
+class FlowProgressIndicator extends Component {
```
Comment (gziolo) [inline client/signup/flow-progress-indicator/index.jsx]: Why not use arrow function style if we no longer have methods here? :)

File: client/signup/config/flows.js [hunk 2]
```diff
@@ -228,7 +228,8 @@ if ( config.isEnabled( 'signup/domain-first-flow' ) ) {
 		destination: getSiteDestination,
 		providesDependenciesInQuery: [ 'siteSlug', 'siteId' ],
 		description: 'A flow to test updating an existing site with `Signup`',
-		lastModified: '2017-01-19'
+		lastModified: '2017-01-19',
```
Comment (gziolo) [inline client/signup/config/flows.js]: I'm wondering if that date should be updated, too. @drewblaisdell @scruffian?

Comment (aidvu) [inline client/signup/config/flows.js]: Prolly yes! :)

**Comment by fabianapsimoes** [comment]
@ranh I wonder if we should also hide the progress indicator in `start/domain-first`. Right now, it's a bit of an odd behavior where you land in step 1 of 4, but if you click "Just buy a domain" you're sent to step 4 of 4.

Apart from that, product :+1:

**Comment by aidvu** [comment]
Hiding it for `domain-first` with this PR is 1 line.

**Comment by gziolo** [comment]
@ranh @fabianapsimoes in the long run it would be much nicear to use percentage based progress indicator. It plays quite well on get.blog :)

**Comment by fabianapsimoes** [comment]
As discussed in p1488991668243294-tortuga-slack, we'll drop this change for now. It makes more sense to leave that as is and then work on better solutions for progress indication (such as the one suggested by @gziolo). Hiding it for some flows and not others will create inconsistent experiences, but also add exceptions we'll likely forget once we start iterating on progress indication.

**Pull request closed without merging**